src/ApiCore/RequestBuilder.php

```diff
@@ -120,7 +120,7 @@ public function build($path, Message $message, array $headers = [])
                     }
 
                     $value = $this->getQuerystringValue($propertyValue);
-                    if ($propertyValue instanceof Message) {
+                    if ($propertyValue instanceof Message && is_array($value)) {
```

Comment:
This is the fix. the special cases reduce `$value` to a scalar, so we can skip the `foreach`